### ISSUE TITLE
conversation.mark_as_read throws PG::ProtocolViolation

### ISSUE THREAD

**Issue by unhinged** [mailboxer/mailboxer#329]
ruby 2.1.4p265 (2014-10-27 revision 48166) [x86_64-darwin14.0]
Rails 4.2.0
psql (PostgreSQL) 9.3.5
mailboxer (0.12.4)

I recently `bundle update`-ed so I'm not sure where I was when everything was working great. However, now I am getting an error when I try to mark a conversation as read...

	convo = Mailboxer::Conversation.find(params[:message_id])
	convo.mark_as_read(current_user)	# <-- throws error

The error message is...

>ActiveRecord::StatementInvalid: PG::ProtocolViolation: ERROR:  bind message supplies 2 parameters, but prepared statement "a15" requires 0
>    : UPDATE "mailboxer_receipts" SET "is_read" = 't' WHERE (id = 9513 )
>    app/controllers/messages_controller.rb:12:in `mark_as_read'

I'm having trouble tracking the error down once it gets into the gem so I'm not sure what else I can say. It seems like a bug with `mailboxer`. I dont think I am doing anything unusual.

**Comment by unhinged** [comment]
I'm wondering if this has anything to do with updating to Rails 4.2 and https://github.com/mailboxer/mailboxer/issues/316#issuecomment-70162968

**Comment by SqueezedLight** [comment]
Any progress on this? I'm facing the exact same problem and can't find a solution....

**Comment by unhinged** [comment]
No. No progress that I am aware of. My project is low priority right now, so I am hoping I can wait it out and when 316 is fixed and published that this would also be addressed too. I don't have any rational reason to believe that that would be true except that both errors occurred after I updated to Rails 4.2. In fact, I cant even reproduce this problem anymore because I am unable recreate and seed my database because of 316.

**Comment by SqueezedLight** [comment]
ok, thanks.

**Comment by hgani** [comment]
I am also getting this problem and am pretty sure this happens after upgrading to Rails 4.2.

@unhinged you can workaround #316 for now using this branch:

gem 'mailboxer', git: 'https://github.com/div/mailboxer.git', branch: 'rails42-foreigner'

**Comment by sungwoncho** [comment]
Thanks @hgani.

**Comment by unhinged** [comment]
I updated to use the `rails42-foreigner` branch and I am receiving this error message again. Still unsure of the solution.

**Comment by hgani** [comment]
@unhinged it's a workaround for issue #316 (foreigner incompatibility), not for the issue in this bug report. As far as I know, there isn't a workaround or fix for this ProtocolViolation issue yet.

**Comment by unhinged** [comment]
@hgani Correct. I understand that. I am reconfirming this issue because I previously reported in this conversation that I stopped receiving the error. I am receiving the error again now that #316 is out of the way.

**Comment by gauravtiwari** [comment]
Any update on this issue yet?

**Comment by rogelio-meza-t** [comment]
what about disabling prepared statements??
Adding this line to `database.yml` works for me

    development:
        [...]
        prepared_statements: false

**Comment by lacco** [comment]
For me this error seems to be fixed with current mailboxer master (this seems to be the revelant commit: https://github.com/mailboxer/mailboxer/commit/cde8b2847a5600256c8aa161f68e22c34603ef8b). When you are on Rails 4.2, use my fork which is based on current master and removes foreigner gem: 

    gem 'mailboxer', github: 'lacco/mailboxer'

**Comment by Jake0oo0** [comment]
Worked for me, thanks @lacco :+1:

**Comment by Seliestel** [comment]
Just had the same problem. I think it's not a problem with rails 4.2, but with the recommended alternative fork to fix the foreigner gem incompatibility. I changed it to the one provided by Iacco and the bug is gone. Thanks @Iacco!

**Comment by gauravtiwari** [comment]
Sorry for late response. Worked for me as well (for rails 4.2 Mailboxer has no issues). :+1:

**Comment by seanpaulkelley** [comment]
this worked for me to resolve PG error but my generated views 
rails g mailboxer:views
are not getting used for sent emails. Looks like this was reported https://github.com/mailboxer/mailboxer/issues/324

I wonder if the generated views works in master now

**Comment by JoaquinSantana** [comment]
big thx @lacco

**Comment by AlfredoRoca** [comment]
Great job @lacco. 
Worked for me as well.

**Comment by kirka121** [comment]
gem 'mailboxer', github: 'lacco/mailboxer' 

gives an error: 'acts_as_messageable no method, for rails 4.2

**Comment by resnate** [comment]
Thanks @lacco :) :+1:
